addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -3383,6 +3383,19 @@ export class Wysiwyg extends Component {
             delete el.dataset.bgSrc;
         } else {
             el.setAttribute('src', newAttachmentSrc);
+            // Also update carousel thumbnail.
+            const carouselEl = el.closest(".carousel");
+            if (carouselEl) {
+                const carouselInnerEl = el.closest(".carousel-inner");
+                const carouselItemEl = el.closest(".carousel-item");
+                if (carouselInnerEl && carouselItemEl) {
+                    const imageIndex = [...carouselInnerEl.children].indexOf(carouselItemEl);
+                    const miniatureEl = carouselEl.querySelector(`.carousel-indicators [data-bs-slide-to="${imageIndex}"]`);
+                    if (miniatureEl) {
+                        miniatureEl.style.backgroundImage = `url(${newAttachmentSrc})`;
+                    }
+                }
+            }
```

Comment:
Why don't you just add the `o_modified_image_to_save` class to the carousel indicator ?

Reply:
I think there is already a smart code that prevents this.
If you upload an image and add it in two different blocks and then save, only one attachment will be created.

Reply:
Indeed...
The `create_unique()` method of the `ir.attachment` model should be fixed then. It should create only if unique even if the calls are done sequentially.

Reply:
I guess you mean "in parallel" rather than "sequentially".
The calls happen in parallel (see https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/js/wysiwyg/wysiwyg.js#L1138) to speed up saving by not waiting for the feedback of every single save before performing the next one.
The `create_unique()` method runs within the Odoo ORM world and cannot perform any magic to traverse SQL transactions.

So, we could queue the JS calls to `/modify_image` to avoid this issue - but that would cost time on low latency networks - for a very unlikely situation, which does not cause any problem anyway (just one extra row in a table that references the same stored binary: we do not consume that storage space twice).

Reply:
Yes we have to keep it in //
We could use a lock psql system on the table but it will be overkill. Can't we just save an array with base64 images we already sent to the `_saveModifiedImage` and don't call it if the current base64 is in the array ?

Reply:
@Guillaume-gdi  Finally tried to tackle this.
So I added a commit to avoid doing several time the same `modify_image` (not pushed). This could even be kept as a distinct improvement.
But then I realised that tagging the thumbnail with the class requires this image-copying code too, but in another location, and with the added complexity of adding the class - because currently the thumbnail source remains the originally uploaded image.
I think this might be even worse. The added value is that during edit, you'd immediately see the change applied on the thumbnail. The down side is that there might be unforeseen ways to mess up the cached image uploads (e.g. various filtering options could lead to the same base64), and that cache might grow until wysiwyg is destroyed, and consume a lot of memory.
So before actually proceeding with this, I prefer to check with you if we really should go for it ?